example/lib/pages/scalebar.dart

```diff
@@ -1,32 +1,28 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_map/flutter_map.dart';
 import 'package:flutter_map_example/misc/tile_providers.dart';
-import 'package:flutter_map_example/plugins/scale_layer_plugin_option.dart';
 import 'package:flutter_map_example/widgets/drawer/menu_drawer.dart';
 import 'package:latlong2/latlong.dart';
 
-class PluginScaleBar extends StatelessWidget {
-  static const String route = '/plugin_scalebar';
+class ScaleBarPage extends StatelessWidget {
+  static const String route = '/scalebar';
 
-  const PluginScaleBar({super.key});
+  const ScaleBarPage({super.key});
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(title: const Text('Scale Bar Plugin')),
-      drawer: const MenuDrawer(PluginScaleBar.route),
+      appBar: AppBar(title: const Text('Scale Bar Layer')),
+      drawer: const MenuDrawer(ScaleBarPage.route),
       body: FlutterMap(
         options: const MapOptions(
           initialCenter: LatLng(51.5, -0.09),
           initialZoom: 5,
         ),
         children: [
           openStreetMapTileLayer,
-          const FlutterMapScaleLayer(
-            lineColor: Colors.black,
-            lineWidth: 3,
-            textStyle: TextStyle(color: Colors.black, fontSize: 14),
-            padding: EdgeInsets.all(10),
+          Scalebar(
+            painter: SimpleScalebarPainter(padding: const EdgeInsets.all(12)),
```

Comment:
There should be some level of abstraction here. The user doesn't need to know that a Painter gets used internally. Padding is probably a parameter that every Scalebar should have therefore provided to the Scalebar widget directly.
Being able to provide an Painter could make it more open on the one side but you prevented an open extendability by the `base` class modifier. Don't get me wrong, I don't think that we should provide this as well but keeping the all Painters internally all together sounds like the better approach to me.

If we want to support multiple Scalebar styles I'd argue that mimicking the way Flutter does it would be good: `Switch()` and `Switch.adaptive()` or `ListView()` and `ListView.builder()`.
 If transfered to our Scalebar: `Scalebar()` would be the simple (no need to use an extra simple here) and further extensions would be created as `Scalebar.dualUnit`, `Scalebar.google` an so on.
This has the advantage that the painter can stay internally but theoretically created already in the constructor and we have full control of the parameters each Scalebar should support.

Reply:
I implemented it like this to allow for users to create their own painter based on the maths in the `Scalebar` widget.
`base` does not prevent external extension, it just forces the use of `extends` rather than `implements`. Can remove the class modifier altogether if you'd prefer that.

I'm also up for changing the name away from `Painter` to give the illusion of abstraction. Maybe `Style`?

Yes, things like `padding` could possibly be moved outside of the painter, but I didn't want to interfere with anything in these commits, just rearrange.

Reply:
Alright let's make it extentable.

Reply:
I looked into this a bit further and came across more problems if we want to let users provide their own painters:
1. The `Scalebar` widget contains not much logic itself, it mostly calculates the width of the scalebar for a metric scale. This means that we can't really have painters based on other measure systems. Providing users access to `_calculateEndingGlobalCoordinates` would give users more flexibility while limiting our public API.
2. The `Scalebar` is build for horizontal sclebars in mind, things would need to get renamed and generalized otherwise.
3. Marking fields as `internal parameter` that shouldn't get used outside isn't good style.
```dart
  /// Internal parameter, do not use externally
  @protected
  abstract double scaleWidth;

  /// Internal parameter, do not use externally
  @protected
  abstract int scaleDistance;
```
4. Most of the times we tell users to build a plugin for similar cases. Wouldn't this be the better solution here too? We can encorage this by open up the calculation from `_calculateEndingGlobalCoordinates`.

Reply:
Yeah, ok, you're right. Feel free to revert my changes, or let me know if you want me to do it.
But I do still think alignment options should be provided.

Reply:
Yes I'll add alignment. 👍